### ISSUE TITLE
custom civ not showing

### ISSUE THREAD

**Issue by ilteroi** [LoneGazebo/Community-Patch-DLL#1252]
just a stupid question, i downloaded JFDs Kingdom of Bavaria, activated it, try to start a game but i cannot find Ludwig. what's up? is this a known issue?

**Comment by LoneGazebo** [comment]
Can't find him?

**Comment by ilteroi** [comment]
well, let me rephrase that: neither the kingdom of bavaria nor it's leader ludwig is available in the game setup screen.

**Comment by LoneGazebo** [comment]
That's real odd. I'd let JFD know.

**Comment by JFDaly** [comment]
ilteroi, database log? This could be a conflict with another mod, or caused by the emergence of the recent and mysterious no customs civs bug. But suffice to say, I can see Ludwig with CPP suite alone.

**Comment by ilteroi** [comment]
database log is here: indeed some bavaria-related things in there:

[1860345.015] columns StrategicViewType, TileType are not unique
[1860345.015] While executing - 'INSERT INTO ArtDefine_StrategicView(StrategicViewType, TileType, Asset) VALUES(?,?,?)'
[1860347.171] no such table: ContentPackage.LocalizedText
[1860347.171] no such table: ContentPackage.LocalizedText
[1860347.171] no such table: ContentPackage.LocalizedText
[1860349.031] no such table: ContentPackage.LocalizedText
[1860351.500] columns StrategicViewType, TileType are not unique
[1860351.500] While executing - 'INSERT INTO ArtDefine_StrategicView(StrategicViewType, TileType, Asset) VALUES(?,?,?)'
[1860352.625] no such table: ContentPackage.LocalizedText
[1860352.625] no such table: ContentPackage.LocalizedText
[1860352.625] no such table: ContentPackage.LocalizedText
[1860358.171] Validating Foreign Key Constraints...
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860358.171] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860361.796] Failed Validation.
[1860362.421] 
-- SQLite Memory Statistics --
Memory Usage:
		[Cur]		[Max]
Malloc:		6223264		75785624
PageCache:	6		12
LookAside:	0		0
Scratch:	0		1

Static Buffer Overflows:
		[TooLarge]	[NoSpace]
PageCache:	5808264		62880720
Scratch:	0		0

Largest Allocations:
Malloc:		262144
PageCache:	1172
Scratch:	6640

Prepared Statements:
Current:		6
------------------------------
[1860362.468] Validating Foreign Key Constraints...
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860362.468] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860365.859] Failed Validation.
[1860366.593] 
-- SQLite Memory Statistics --
Memory Usage:
		[Cur]		[Max]
Malloc:		6224440		75785624
PageCache:	6		12
LookAside:	0		0
Scratch:	0		1

Static Buffer Overflows:
		[TooLarge]	[NoSpace]
PageCache:	5809440		62880720
Scratch:	0		0

Largest Allocations:
Malloc:		262144
PageCache:	1172
Scratch:	6640

Prepared Statements:
Current:		6
------------------------------
[1860454.906] no such column: CivilizationType
[1860457.187] Validating Foreign Key Constraints...
[1860457.187] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.187] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.203] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.203] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.203] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.203] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.203] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.203] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860457.203] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860459.921] Invalid Reference on Improvements.CivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860462.468] Invalid Reference on Leader_SharedTraits.TraitTwo - "POLICY_INDUSTRIOUS_X" does not exist in Policies
[1860462.468] Invalid Reference on Leader_SharedTraits.TraitOne - "POLICY_CHARISMATIC_X" does not exist in Policies
[1860462.468] Invalid Reference on Civilization_JFD_CultureTypes.CivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860462.468] Invalid Reference on Civilization_JFD_ColonialCityNames.CivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860462.468] Invalid Reference on Civilization_JFD_RevolutionaryCivilizationsToSpawn.RevolutionaryCivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860462.468] Failed Validation.
[1860463.906] 
-- SQLite Memory Statistics --
Memory Usage:
		[Cur]		[Max]
Malloc:		8438696		75785624
PageCache:	8		14
LookAside:	0		0
Scratch:	0		1

Static Buffer Overflows:
		[TooLarge]	[NoSpace]
PageCache:	7792176		62880720
Scratch:	0		0

Largest Allocations:
Malloc:		262144
PageCache:	1172
Scratch:	6640

Prepared Statements:
Current:		18
------------------------------
[1860464.031] Validating Foreign Key Constraints...
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860464.031] Invalid Reference on ArtDefine_Landmarks.LayoutHandler - "SPECIAL" does not exist in ArtDefine_LandmarkTypes
[1860465.921] Invalid Reference on Improvements.CivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860467.968] Invalid Reference on Leader_SharedTraits.TraitTwo - "POLICY_INDUSTRIOUS_X" does not exist in Policies
[1860467.968] Invalid Reference on Leader_SharedTraits.TraitOne - "POLICY_CHARISMATIC_X" does not exist in Policies
[1860467.968] Invalid Reference on Civilization_JFD_CultureTypes.CivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860467.968] Invalid Reference on Civilization_JFD_ColonialCityNames.CivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860467.968] Invalid Reference on Civilization_JFD_RevolutionaryCivilizationsToSpawn.RevolutionaryCivilizationType - "CIVILIZATION_JFD_BAVARIA" does not exist in Civilizations
[1860467.968] Failed Validation.
[1860469.453] 
-- SQLite Memory Statistics --
Memory Usage:
		[Cur]		[Max]
Malloc:		8439872		75785624
PageCache:	8		14
LookAside:	0		0
Scratch:	0		1

Static Buffer Overflows:
		[TooLarge]	[NoSpace]
PageCache:	7793352		62880720
Scratch:	0		0

Largest Allocations:
Malloc:		262144
PageCache:	1172
Scratch:	6640

Prepared Statements:
Current:		18
------------------------------
[1860497.125] no such column: Type
[1860497.125] In Query - select * from Natural_Wonder_Placement where Type = ? LIMIT 1

**Comment by JFDaly** [comment]
Yup, that's indeed the mysterious bug. Please check the Remapper.sql file in (2) Community Balance Patch\Core Files. There shouldn't be a trigger called 'CBP_Civilization_UnitClassOverrides' at the bottom (because it was removed in Feb. 10th version), but it's my only - albeit weak - lead on this issue. If it's there, delete it.

**Comment by ilteroi** [comment]
yup, that did the trick. thanks!

it's a bit strange though because i am on the 2/10 version

**Comment by LoneGazebo** [comment]
I figured it might be the trigger. I'm removing it in the next version anyways, so we can consider this solved.
